### PR TITLE
fix(weave): Move metadata into its own field to avoid collisions with other pkgs

### PR DESCRIPTION
**Pull request by andrewtruong**
Resolves:
- https://wandb.atlassian.net/browse/WB-19388

This PR moves metadata that was previously top level (`name`, `description`) to a new top-level `metadata` field which prevents collision with other libraries that also have `name` as a top-level field (e.g. `openai`)

### CODE REVIEW

**Review by tssweeney — APPROVED** [review]
Primary comment is that this should be under a different namespace - maybe `weave_meta` or something like that. Perhaps even underscored

File: weave/tests/test_graph_client.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import queue
```
Comment (tssweeney) [inline weave/tests/test_graph_client.py:1]: accidentally included?

File: weave/tests/test_weave_client.py [hunk 2]
```diff
@@ -288,7 +288,7 @@ async def score(target, model_output):
         return target == model_output
 
     evaluation = Evaluation(
-        name="my-eval",
+        metadata={"name": "my-eval"},
```
Comment (tssweeney) [inline weave/tests/test_weave_client.py:291]: this feels very odd to me - does it feel odd to you?

Comment (andrewtruong) [reply weave/tests/test_weave_client.py:291]: Why?

File: weave/tests/test_graph_client.py (hunk 1 above)
Comment (andrewtruong) [reply weave/tests/test_graph_client.py:1]: this branch is super outdated

**Review by tssweeney — CHANGES_REQUESTED** [review]
Remove unused code before landing

File: weave/tests/test_weave_client.py (hunk 2 above)
Comment (tssweeney) [reply weave/tests/test_weave_client.py:291]: well, as a user, i am thinking to myself: i want to name this evaluation, why is it some nested dictionary? It feels like the underlying data model is leaking into user-land. If we didn't have this name collision issue, we would not choose this API. There might not be a good alt, just want to discuss / see if there are other ideas

Comment (andrewtruong) [reply weave/tests/test_weave_client.py:291]: It feels right to me.  This dict format is just taking advantage of pydantic so you don't need to use the class.  You could alternatively write:
```python
metadata=Metadata(name="my-eval")
```

That said, I wonder if having a very general metadata dict is also useful -- basically simple k,v tags for objects

Comment (andrewtruong) [reply weave/tests/test_weave_client.py:291]: For `Evaluation` specifically, we could also have `name` as the top-level then forward the value to `metadata`

**Pull request closed without merging**